fix(sql): fix tables partitioning with mysql 8

sc979 opened the pull request:
# Pull Request Template

## Description

On mysql > 5.9, the plugin “partition” as been deprecated and removed.
Resulting on the failling upgrade/install process, as the Cron job.
We need to adapt our code to remove the plugin presence on mysql > 5.9

`https://docs.oracle.com/cd/E17952_01/mysql-8.0-en/faqs-storage-engines.html`

**Fixes** # (none)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).
- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

kduret commented on www/class/centreon-partition/partEngine.class.php:
```diff
@@ -284,34 +283,34 @@ public function createParts($table, $db, $createPastPartitions): void
 
         // maybe we need to check the value of $row["lastPart"]
-        return($row["lastPart"]);
+        return ($row["lastPart"]);
```
```suggestion
        return $row["lastPart"];
```

kduret commented on www/class/centreon-partition/partEngine.class.php:
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
+                    case 'version' :
+                        $dbVersion = $row['Value'];
+                        break;
+                    default :
```
remove default case

kduret commented on www/class/centreon-partition/partEngine.class.php:
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
+            $dbType = $dbVersion = null;
+            while ($row = $dbResult->fetch()) {
+                switch ($row['Variable_name']) {
+                    case 'version_comment' :
```
```suggestion
                    case 'version_comment':
```

kduret commented on www/class/centreon-partition/partEngine.class.php:
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
+                    case 'version_comment' :
+                        $dbType = $row['Value'];
+                        break;
+                    case 'version' :
```
```suggestion
                    case 'version':
```

sc979 commented:
rebased on each specific branches and added to these PR :
https://github.com/centreon/centreon/pull/7801
https://github.com/centreon/centreon/pull/7799
https://github.com/centreon/centreon/pull/7798
